db/seeds/partials/_languages.rb

```diff
@@ -0,0 +1,943 @@
+    "name": "Somalia",
+    "native": "Soomaaliga"
+  },
+  {
+    "code": "sq",
+    "name": "Albanian",
+    "native": "Shqip"
+  },
+  {
+    "code": "sr",
+    "name": "Serbian",
+    "native": "Српски"
+  },
+  {
+    "code": "ss",
+    "name": "Swati",
+    "native": "SiSwati"
+  },
+  {
+    "code": "st",
+    "name": "Southern Sotho",
+    "native": "Sesotho"
+  },
+  {
+    "code": "su",
+    "name": "Sundanese",
+    "native": "Basa Sunda"
+  },
+  {
+    "code": "sv",
+    "name": "Swedish",
+    "native": "Svenska"
+  },
+  {
+    "code": "sw",
+    "name": "Swahili",
+    "native": "Kiswahili"
+  },
+  {
+    "code": "ta",
+    "name": "Tamil",
+    "native": "தமிழ்"
+  },
+  {
+    "code": "te",
+    "name": "Telugu",
+    "native": "తెలుగు"
+  },
+  {
+    "code": "tg",
+    "name": "Tajik",
+    "native": "Тоҷикӣ"
+  },
+  {
+    "code": "th",
+    "name": "Thai",
+    "native": "ไทย / Phasa Thai"
+  },
+  {
+    "code": "ti",
+    "name": "Tigrinya",
+    "native": "ትግርኛ"
+  },
+  {
+    "code": "tk",
+    "name": "Turkmen",
+    "native": "Туркмен / تركمن"
+  },
+  {
+    "code": "tl",
+    "name": "Tagalog / Filipino",
+    "native": "Tagalog"
+  },
+  {
+    "code": "tn",
+    "name": "Tswana",
+    "native": "Setswana"
+  },
+  {
+    "code": "to",
+    "name": "Tonga",
+    "native": "Lea Faka-Tonga"
+  },
+  {
+    "code": "tr",
+    "name": "Turkish",
+    "native": "Türkçe"
+  },
+  {
+    "code": "ts",
+    "name": "Tsonga",
+    "native": "Xitsonga"
+  },
+  {
+    "code": "tt",
+    "name": "Tatar",
+    "native": "Tatarça"
+  },
+  {
+    "code": "tw",
+    "name": "Twi",
+    "native": "Twi"
+  },
+  {
+    "code": "ty",
+    "name": "Tahitian",
+    "native": "Reo Mā`ohi"
+  },
+  {
+    "code": "ug",
+    "name": "Uyghur",
+    "native": "Uyƣurqə / ئۇيغۇرچە"
+  },
+  {
+    "code": "uk",
+    "name": "Ukrainian",
+    "native": "Українська"
+  },
+  {
+    "code": "ur",
+    "name": "Urdu",
+    "native": "اردو",
+    "rtl": 1
+  },
+  {
+    "code": "uz",
+    "name": "Uzbek",
+    "native": "Ўзбек"
+  },
+  {
+    "code": "ve",
+    "name": "Venda",
+    "native": "Tshivenḓa"
+  },
+  {
+    "code": "vi",
+    "name": "Vietnamese",
+    "native": "Tiếng Việt"
+  },
+  {
+    "code": "vo",
+    "name": "Volapük",
+    "native": "Volapük"
+  },
+  {
+    "code": "wa",
+    "name": "Walloon",
+    "native": "Walon"
+  },
+  {
+    "code": "wo",
+    "name": "Wolof",
+    "native": "Wollof"
+  },
+  {
+    "code": "xh",
+    "name": "Xhosa",
+    "native": "isiXhosa"
+  },
+  {
+    "code": "yi",
+    "name": "Yiddish",
+    "native": "ייִדיש",
+    "rtl": 1
+  },
+  {
+    "code": "yo",
+    "name": "Yoruba",
+    "native": "Yorùbá"
+  },
+  {
+    "code": "za",
+    "name": "Zhuang",
+    "native": "Cuengh / Tôô / 壮语"
+  },
+  {
+    "code": "zh",
+    "name": "Chinese",
+    "native": "中文"
+  },
+  {
+    "code": "zu",
+    "name": "Zulu",
+    "native": "isiZulu"
+  }
+]
+
+langs.each do |lang|
```

Comment:
Instead of using create many times to insert one language at a time into the database, it is (much, like 9 times) faster to create a single SQL insert statement and send them all in at once. If you have your data as an array of hashes like you do here, you can use `upsert_all`, like this:

`Language.upsert_all langs`

I did this in the `_challenges.rb` seed file if you want to have a look for reference.
I'm just not 100% that it works with strings as keys, so please test first.

Reply:
Ah ok, I saw you use it last week and didn't think to use it here thanks